tests/components/bmw_connected_drive/test_sensor.py

```diff
@@ -31,21 +34,50 @@ async def test_entity_state_attrs(
     assert hass.states.async_all("sensor") == snapshot
```

Comment:
Please use the snapshot_platform helper. It will also give a right name to the platform, so if you change something (add a sensor, or rename one), it doesn't completely mess up the git diff (as this is a list, and the snapshot_platform helper makes it more like a dict.

Maybe it's a good idea to migrate to the helper in a preliminary PR

Reply:
Thanks, that makes sense!

Added it here as well as in https://github.com/home-assistant/core/pull/118436 and https://github.com/home-assistant/core/pull/118735.
Depending on what gets merged when, will rebase/fix merge conflicts as they come up.